usagov/usagov-2021
# USAGOV-644-move-states-out-cms: Move states out of cms with hook

## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-644

## Description

Appended to hook usagov_theme_suggestions_alter found in usagov.theme to allow us to create twig templates for code that was in the body field in the cms. Essentially this hook gets the current pages alias and appends it to the string 'field__node__body__basic_page_'. The new string is then added as a twig template suggestion.

Item(s) of note:

- This is based off the work of [USAGOV-646](https://github.com/usagov/usagov-2021/pull/623/files). That work's PR has been converted to a draft as the actually content & code to be moved has to be worked out. This page only has a form so the entire code was able to be moved out.

- For some reason the string in the theme file has to use underscores NOT dashes. The actual twig template can use dashes but it took hours of debugging to find that Drupal will only find the template if the hook uses underscores.

- A previous implementation of this used an include statement in the field--node--basic_page. Though this allowed the English and Spanish templates to be in the same file (which this implementation does not) it lead to a cluttered basic_page template.

- For the hook to be called the "Custom Twig Content" checkbox has to be selected in the CMS

### Testing
- Import config changes (bin/drush cim)
- Go to localhost/states
- Select Edit -> Remove code within the body field
- Check that the drop down is gone 
- Select Edit -> "Custom Twig Content" 
- Check that the drop down is back and works 

## Checklist for the Developer



## Checklist for the Peer Reviewers
